File: spring-desafio-backend/src/main/java/net/javaguides/springboot/service/UsuarioService.java
package net.javaguides.springboot.service;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;

import net.javaguides.springboot.exception.ResourceNotFoundException;
import net.javaguides.springboot.model.Usuario;
import net.javaguides.springboot.repository.UsuarioRepository;

@Service
public class UsuarioService {
	
	@Autowired
	private UsuarioRepository usuarioRepository;
	
	
	public List<Usuario> listaTodos(){
		List<Usuario> users = usuarioRepository.findAll();
		return users;
	}
	
	public Usuario criarUsuario(@RequestBody Usuario usuario) {
		Usuario user = usuarioRepository.save(usuario);
		return user;
	}
	
	public ResponseEntity<Usuario> buscaUsuarioPorId(@PathVariable Long id) {
		Usuario usuario = usuarioRepository.findById(id)
				.orElseThrow(() -> new ResourceNotFoundException("Não existe usuario com o id :" + id));
		return ResponseEntity.ok(usuario);
	}
	
	public ResponseEntity<Usuario> atualizarUsuario(@PathVariable Long id, @RequestBody Usuario usuarioDetalhes){
		Usuario usuario = usuarioRepository.findById(id)
				.orElseThrow(() -> new ResourceNotFoundException("Não existe usuario com o id :" + id));
		
		usuario.setNome(usuarioDetalhes.getNome());
		usuario.setIdade(usuarioDetalhes.getIdade());
		usuario.setSkill(usuarioDetalhes.getSkill());
		
		Usuario atualizarUsuario = usuarioRepository.save(usuario);
		return ResponseEntity.ok(atualizarUsuario);
	}
	
	public ResponseEntity<Map<String, Boolean>> deletarUsuario(@PathVariable Long id){
		Usuario usuario = usuarioRepository.findById(id)
				.orElseThrow(() -> new ResourceNotFoundException("Não existe usuario com o id :" + id));
		
		usuarioRepository.delete(usuario);
		Map<String, Boolean> response = new HashMap<>();
		response.put("deletado", Boolean.TRUE);
		return ResponseEntity.ok(response);
	}

}

File: spring-react-desafio-frontend/react-frontend/src/services/UsuarioService.js
import axios from 'axios';

const USUARIO_API_BASE_URL = "http://localhost:8080/api/v1/usuarios";

class UsuarioService {

    listaTodos(){
        return axios.get(USUARIO_API_BASE_URL);
    }

    criarUsuario(usuario){
        return axios.post(USUARIO_API_BASE_URL, usuario);
    }

    buscaUsuarioPorId(usuarioId){
        return axios.get(USUARIO_API_BASE_URL + '/' + usuarioId);
    }

    atualizarUsuario(usuario, usuarioId){
        return axios.put(USUARIO_API_BASE_URL + '/' + usuarioId, usuario);
    }

    deletarUsuario(usuarioId){
        return axios.delete(USUARIO_API_BASE_URL + '/' + usuarioId);
    }
}

export default new UsuarioService()